shell/platform/fuchsia/flutter/platform_view.cc

```diff
@@ -96,19 +76,18 @@ PlatformView::PlatformView(
       await_vsync_for_secondary_callback_callback_(
           await_vsync_for_secondary_callback_callback),
       weak_factory_(this) {
-  // Register all error handlers.
-  SetInterfaceErrorHandler(ime_, "Input Method Editor");
-  SetInterfaceErrorHandler(ime_client_, "IME Client");
-  SetInterfaceErrorHandler(text_sync_service_, "Text Sync Service");
-  SetInterfaceErrorHandler(keyboard_listener_binding_, "Keyboard Listener");
-  SetInterfaceErrorHandler(keyboard_, "Keyboard");
-
   fuchsia::ui::views::ViewRef view_ref_clone;
   fidl::Clone(view_ref, &view_ref_clone);
 
-  // Configure keyboard listener.
-  keyboard_->AddListener(std::move(view_ref),
-                         keyboard_listener_binding_.NewBinding(), [] {});
+  text_delegate_ = std::make_unique<TextDelegate>(
+      std::move(view_ref), std::move(ime_service), std::move(keyboard),
+      // Assumes that |delegate| lives long enough for this closure to
+      // call it. This should be the case, given that |delegate| is
+      // passed into |PlatformView| as a reference, which assumes it
+      // outlives |PlatformView|, and thus also |text_delegate_|.
+      [&delegate](std::unique_ptr<flutter::PlatformMessage> message) {
```

Comment:
It's still the same object "this" getting referenced. Perhaps capture the weak ptr of "this" (like the focus delegate watch loop), in which case, you can say weak->delegate_.On...(); without lifecycle worries.

Reply:
Done.